tests/unit/test_io_hdf5_h5tools.py

```diff
@@ -265,6 +265,22 @@ def test_write_dataset_iterable_multidimensional_array(self):
         dset = self.f['test_dataset']
         self.assertListEqual(dset[:].tolist(), a.tolist())
 
+    def test_write_multi_dci(self):
+        a = np.arange(30).reshape(5, 2, 3)
+        b = np.arange(30, 60).reshape(5, 2, 3)
+        aiter = iter(a)
+        biter = iter(b)
+        daiter1 = DataChunkIterator.from_iterable(aiter, buffer_size=2)
+        daiter2 = DataChunkIterator.from_iterable(biter, buffer_size=2)
+        builder = GroupBuilder("root")
+        builder.add_dataset('test_dataset1', daiter1, attributes={})
+        builder.add_dataset('test_dataset2', daiter2, attributes={})
+        self.io.write_builder(builder)
+        dset1 = self.f['test_dataset1']
```

Comment:
Should have this test be run twice, i.e., once fro  exhaust_dci=False and once with exhaust_dci=True?